### PR TITLE
Enable ruff pydocstyle linter rule

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Enable pydocstyle again, but now with ruff.

Related to #2528 

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: tests/integration/preprocessor/_regrid/test_extract_coordinate_points.py [hunk 1]
```diff
@@ -1,8 +1,4 @@
-"""
-Integration tests for the :func:`esmvalcore.preprocessor.regrid.regrid`
-function.
-
-"""
+"""Integration tests for the :func:`esmvalcore.preprocessor.regrid`."""
```
Comment (schlunma) [inline tests/integration/preprocessor/_regrid/test_extract_coordinate_points.py:1]:
```suggestion
"""Integration tests for :func:`esmvalcore.preprocessor.regrid`."""
```

File: tests/integration/preprocessor/_regrid/test_extract_point.py [hunk 2]
```diff
@@ -1,8 +1,4 @@
-"""
-Integration tests for the :func:`esmvalcore.preprocessor.regrid.regrid`
-function.
-
-"""
+"""Integration tests for the :func:`esmvalcore.preprocessor.regrid`."""
```
Comment (schlunma) [inline tests/integration/preprocessor/_regrid/test_extract_point.py:1]:
```suggestion
"""Integration tests for :func:`esmvalcore.preprocessor.regrid`."""
```

**Review by schlunma — APPROVED** [review]
Great, thanks! I only have some tiny comments.

**Comment by valeriupredoi** [comment]
sorry @bouweandela I messed up your clean merge by merging #2536 first - could you pls fix the conflict? Merge at will after, I will approve now :beer:

**Review by valeriupredoi — APPROVED** [review]
actually, maybe address me comments in #2540 first, then merge that into `main`, then come back here - bet there'll be conflicts again in pyproject.toml

**Pull request merged**